src/jit/codegenxarch.cpp

```diff
@@ -757,7 +757,7 @@ void CodeGen::genCodeForBinary(GenTree* treeNode)
     }
     // now we know there are 3 different operands so attempt to use LEA
     else if (oper == GT_ADD && !varTypeIsFloating(treeNode) && !treeNode->gtOverflowEx() // LEA does not set flags
-             && (op2->isContainedIntOrIImmed() || !op2->isContained()))
+             && (op2->isContainedIntOrIImmed() || !op2->isContained()) && !treeNode->gtSetFlags())
```

Comment:
I see. Codegen's `lea` generation deals with the `targetReg` != `srcReg1` case and if we don't use `lea` we'll probably get an extra useless `mov`. Not ideal but cheaper than a `test`. Lower's `lea` generation takes care of more complicated cases and it may be a good thing that it's not affected by this.